### PR TITLE
Tutorial Levels arranged into a 2D structure

### PR DESCRIPTION
**Pull request by H3rby7**
looks like this: 
```js
tutorials = [
  {
    categoryName: 'sampleCategory',
    levels: [
      require('./someLevel.js'),
      require('./someOtherLevel.js')
    ]
  }
]
``` 

This changes the routing for tutorials again: `localhost:9966/#tutorial/CATEGORY/LEVEL`. Levels start with index 1.

Would be nice for the Overview page something like this:

```html
<h1>Category</h1>
<div class="levels">
  <div style="display: inline-block">
    <img src="linkToImageOfLevel1">
  </div>
  <div style="display: inline-block">
    <img src="linkToImageOfLevel2">
  </div>
  <div style="display: inline-block">
    <img src="linkToImageOfLevel3">
  </div>
</div>
```

### CODE REVIEW

File: elements/tutorial/tutorialDialog.js [hunk 1]
```diff
@@ -147,16 +147,11 @@ const winningCondition = (gameState, { level, isStoryModalOpen }, workspace, sen
...
 function getNextLevelButton (send, level) {
-  if (level.nextTutorial) {
-    return buttonView({
-      label: 'Next Level',
-      onClick: () => send('location:set', `#tutorial/${level.nextTutorial}`)
-    })
-  }
-
   return buttonView({
-    label: 'Load Editor',
```
Comment (timgrossmann) [inline elements/tutorial/tutorialDialog.js:158]: Why did you remove the Load Editor Button, shouldn't we need that for the last tutorial ?

File: models/tutorial/index.js [hunk 2]
```diff
@@ -36,18 +36,40 @@ module.exports = {
...
+        currentLevel = { category: category, index: parseInt(index) }
         send('tutorial:resetEvents', null, _.noop)
         send('tutorial:_setLevel', { level }, _.noop)
         send('game:loadGameState', { loadState: level.game }, _.noop)
         send('editor:update', { workspace: level.editor.workspace }, _.noop)
       }
     },
 
+    nextLevel: (state, cb, send) => {
+      if (!currentLevel || !currentLevel.hasOwnProperty('category') || !currentLevel.hasOwnProperty('index')) return
```
Comment (timgrossmann) [inline models/tutorial/index.js]: This looks nasty... Especially having the return inline

File: models/tutorial/index.js [hunk 3]
```diff
@@ -36,18 +36,40 @@ module.exports = {
...
+        console.log('loading next level of same category')
+        return cb('#tutorial/' + prevCategory + '/' + (prevIndex + 1))
+      }
+      const oldCategoryIndex = _.findIndex(levels, { 'categoryName': currentLevel.category })
+      if (oldCategoryIndex === -1 || oldCategoryIndex + 1 === levels.length) {
+        // was last category or unknown
+        return cb('#editor')
+      }
+      const nextCategory = levels[oldCategoryIndex + 1]
+      if (!nextCategory || !nextCategory.hasOwnProperty('categoryName')) return cb('')
```
Comment (timgrossmann) [inline models/tutorial/index.js]: Again 😢

File: models/tutorial/index.js [hunk 4]
```diff
@@ -82,3 +104,19 @@ function handleClockEvent (events, event, send) {
     })
   }
 }
+
+function getLevelIfExistent (category, index) {
```
Comment (timgrossmann) [inline models/tutorial/index.js:119]: I know we only have 1 more day left, but I'm not sure if we should leave this function like that

**Comment by timgrossmann** [comment]
@H3rby7 Could be please not use inline If-Conditions at all?

**Comment by timgrossmann** [comment]
To keep it consistent, i would recommend changing 'ID' to 'Id' in e.g. tileId.

File: elements/tutorial/tutorialDialog.js (hunk 1 above)
Comment (H3rby7) [reply elements/tutorial/tutorialDialog.js:158]: If there are no more levels present it will jump to the editor.

File: models/tutorial/index.js (hunk 4 above)
Comment (H3rby7) [reply models/tutorial/index.js:119]: Could you specify what you mean and give a suggestion? Maybe we can change this after tomorrow if it is a code style issue.

File: models/tutorial/index.js (hunk 2 above)
Comment (H3rby7) [reply models/tutorial/index.js]: I am open to suggestions 👍
